frontend/app/work_packages/directives/inplace_editor/custom/editable/inplace-editor-date-directive.js

```diff
@@ -0,0 +1,158 @@
+      scope.fieldController = fieldController;
+      var timerId = 0,
+          prevDate = '',
+          input = element,
+          addClearButton = function (inp) {
+            setTimeout(function() {
+              var buttonPane = jQuery('.ui-datepicker-buttonpane');
+
+              if(buttonPane.find('.ui-datepicker-clear').length === 0) {
+                buttonPane.find('.ui-datepicker-clear').remove();
+              }
+
+              jQuery( '<button>', {
+                    text: 'Clear',
+                    click: function() {
+                      inp.val('');
+                      inp.change();
+                    }
+                })
+                .addClass('ui-datepicker-clear ui-state-default ui-priority-primary ui-corner-all')
+                .appendTo(buttonPane);
+            }, 1);
+          },
+          setDate = function(inp, date) {
+            if(date) {
+              inp.datepicker('option', 'defaultDate', formattedDate(date));
+              inp.datepicker('option', 'setDate', formattedDate(date));
+              inp.val(formattedDate(date));
+            } else {
+              inp.datepicker('option', 'defaultDate', null);
+              inp.datepicker('option', 'setDate', null);
+              inp.val('');
+              date = null;
+            }
+          };
+
+      input.attr('placeholder', '');
+
+      input.on('change', function() {
+        if(input.val().replace(/^\s+|\s+$/g, '') === '') {
+          $timeout(function() {
+            scope.fieldController.writeValue = null;
+          });
+          input.val('');
+          $timeout.cancel(timerId);
+          return;
+        }
+        $timeout.cancel(timerId);
+        timerId = $timeout(function() {
+          var date = input.val(),
+              isValid = TimezoneService.isValid(date, 'DD/MM/YYYY');
+
+          if(isValid){
+            scope.fieldController.writeValue = formattedISODate(date);
+          }
+        }, 1000);
+      });
+
+      input.datepicker({
+        firstDay: ConfigurationService.startOfWeek(),
+        showWeeks: true,
+        changeMonth: true,
+        numberOfMonths: 1,
+        dateFormat: 'dd/mm/yy',
+        alterOffset: function(offset) {
+          var wHeight = jQuery(window).height(),
+              dpHeight = jQuery('#ui-datepicker-div').height(),
+              inputTop = input.offset().top,
+              inputHeight = input.innerHeight();
+
+          if((inputTop + inputHeight + dpHeight) > wHeight) {
+            offset.top -= inputHeight - 4;
+          }
+          return offset;
+        },
+        beforeShow: function() {
+          addClearButton(input);
+        },
+        onChangeMonthYear: function() {
+          addClearButton(input);
+        },
+        onClose: function(selectedDate) {
+          if(!selectedDate || selectedDate === '' || selectedDate === prevDate) {
+            return;
+          }
+          var parsedDate = parseDate(selectedDate, 'DD/MM/YYYY');
```

Comment:
I've spotted the `DD/MM/YYYY` constant three times in this file.

Leading to two questions:
1. Why is it not a common constant used in all three places?
2. Why is it a constant at all and not defined by the `ConfigurationService.dateFormat`?